### PR TITLE
Add more integration tests

### PR DESCRIPTION
**Pull request by qleisan**
Add int-102, int-105, int-107, int-301, int-302

### CODE REVIEW

**Review by mlasch — CHANGES_REQUESTED** [review]

File: tests/integration/test_information_reporting_interface.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    assert lwm2mserver.commandresponse("time 0 /3/0/7 5 15", "OK")
+    assert lwm2mserver.commandresponse("time 0 /3/0/8 10 20", "OK")
+    # Test Procedure 3
+    assert lwm2mserver.commandresponse("observe 0 /3/0/7", "OK")
+    assert lwm2mserver.commandresponse("observe 0 /3/0/8", "OK") #consumes first observe from /3/0/7
+    text = lwm2mserver.waitforpacket() # consumes ACK and piggy-backed first observe from /3/0/8
+    text = lwm2mserver.waitfortime(20) # wait for max "pmax"
+    # Pass-Criteria A
+    assert text.count("Notify from client #0 /3/0/7") > 0
+    assert text.count("Notify from client #0 /3/0/8") > 0
```
Comment (mlasch) [inline tests/integration/test_information_reporting_interface.py:31]: Would it be feasible to check here also the values of Power Source Voltage (7) and Power Source Current (8)? Would be nice to check if the notification messages actually contain the observed values.

File: tests/integration/test_registration_interface.py [hunk 2]
```diff
@@ -42,3 +43,87 @@ def test_registration_interface(lwm2mserver, lwm2mclient):
...
+    assert text.find("COAP_204_CHANGED") > 0
+    # b) NOT IMPLEMENTED
+    # c) not observable
+    # Test Procedure 3
+    text = lwm2mserver.waitfortime(120)
+    # Pass-Criteria C
+    # one update is sent immediately, expect implementation
+    # to send update at lifetime/2 intervals
+    # this does not seem to be the case for 120s (more frequent)
+    assert text.count("Client #0 updated.") >= 2
```
Comment (mlasch) [inline tests/integration/test_registration_interface.py:129]: Why not test for exactly two updates during the lifetime? This however interferes with #613 when testing against 120s.

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -46,6 +48,13 @@ def waitfortext(self,txt):
             return False
         return True
 
+    def waitfortime(self, timedelay):
+        '''Wait for time and return output since last command'''
```
[the remark this answers is not in the document]
Comment (qleisan) [reply tests/integration/conftest.py:52]: will fix

File: tests/integration/test_information_reporting_interface.py (hunk 1 above)
Comment (qleisan) [reply tests/integration/test_information_reporting_interface.py:31]: Will add payload check

File: tests/integration/test_registration_interface.py [hunk 4]
```diff
@@ -42,3 +43,87 @@ def test_registration_interface(lwm2mserver, lwm2mclient):
...
+    # wait long enough to client to send register update (180s total)
+    # Pass-Criteria C
+    # only in debug build it is possible to detect "4.04" from printouts
+    # Pass-Criteria D
+    # Client should perform full registration according as in int-101
+    # this is not implemented, client exits with "STATE_BOOTSTRAP_REQUIRED"
+
+
+@pytest.mark.slow
+#@pytest.mark.client_args("-t 60")
```
[the remark this answers is not in the document]
Comment (qleisan) [reply tests/integration/test_registration_interface.py:105]: The entire testcase needs to be removed due to #607

File: tests/integration/test_registration_interface.py (hunk 2 above)
Comment (qleisan) [reply tests/integration/test_registration_interface.py:129]: TC removed. (The exact number of updates is implementation dependent)

**Review by mlasch — APPROVED** [review]

**Pull request merged**